gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -492,9 +494,25 @@ public GradeSaveResponse saveGrade(final Long assignmentId, final String student
 
 			// only convert if we had a previous value otherwise it will be out of sync
 			if (StringUtils.isNotBlank(oldGradeAdjusted)) {
+				// To check if our data is out of date, we first compare what we think
+				// is the latest saved score against score stored in the database. As the score
+				// is stored as points, we must convert this to a percentage. To be sure we're
+				// comparing apples with apples, we first determine the number of decimal places
+				// on the score, so the converted points-as-percentage is in the expected format. 
+				int numberOfDecimalPlaces = 0;
+				if (storedGradeAdjusted.indexOf(".") >= 0) {
+					numberOfDecimalPlaces = storedGradeAdjusted.split("\\.")[1].length();
+				}
+
 				final Double oldGradePercentage = NumberUtils.toDouble(oldGrade);
 				final Double oldGradePointsFromPercentage = (oldGradePercentage / 100) * maxPoints;
-				oldGradeAdjusted = FormatHelper.formatDoubleToTwoDecimalPlaces(oldGradePointsFromPercentage);
```

Comment:
Thanks @steveswinsburg.  I've refactored it to move the munging into the `FormatHelper`.  Let me know what you think.